Sustainsys.Saml2/SAML2P/Saml2PSecurityTokenHandler.cs

```diff
@@ -17,10 +17,15 @@ namespace Sustainsys.Saml2.Saml2P
 	/// could be handled at transport level.
 	/// </summary>
 	public class Saml2PSecurityTokenHandler : Saml2SecurityTokenHandler
-    {
-		public Saml2PSecurityTokenHandler()
+	{
+		public Saml2PSecurityTokenHandler(SPOptions spOptions)
 		{
-			Serializer = new Saml2PSerializer();
+			if (spOptions == null)
+			{
+				throw new ArgumentNullException(nameof(spOptions));
```

Comment:
Missing test coverage on this line - but let's figure out how to deal with the change of signature first.